.github/workflows/publish_release.yml

```diff
@@ -1,79 +1,61 @@
 name: Publish a new release
 
 on:
-  workflow_dispatch:
-    inputs:
-      is_major:
-        type: boolean
-        required: false
-        description: Make a major version
-      breaking_changes:
-        type: string
-        required: false
-        description: Breaking changes message
+  push:
+    branches:
+    - main
+    
 
 jobs:
   build:
-    name: Build
+    name: Build candy-doc
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
       uses: actions/checkout@v3
-
+    - name: Import GPG key
+      id: import_gpg
+      uses: crazy-max/ghaction-import-gpg@v4
+      with:
+        gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+        passphrase: ${{ secrets.GPG_PASSPHRASE }}
     - name: Set up JDK 11
       uses: actions/setup-java@v3
       with:
         java-version: '11'
         distribution: adopt
         cache: maven
-
     - name: Build with Maven
       run: mvn --batch-mode -DskipTests package --update-snapshots verify
 
-
-  prepare-major:
-    name: Prepare for a new major version
+  release:
+    name: Semantic-Release
     runs-on: ubuntu-latest
-    outputs:
-      output1: ${{ steps.get-last-commit-ref.outputs.ref }}
-    if: ${{ github.event.inputs.isMajor == 'true' }}
+    needs: build
     steps:
     - name: Checkout
       uses: actions/checkout@v3
+      
+    - name: Import GPG key
+      id: import_gpg
+      uses: crazy-max/ghaction-import-gpg@v4
       with:
-        fetch-depth: 2
-
-    - name: Get last commit message
-      id: last-commit-message
-      run: |
-        echo "::set-output name=msg::$(git log -1 --pretty=%s)"
-
-    - uses: stefanzweifel/git-auto-commit-action@v4
-      with:
-        commit_message: "${{ steps.last-commit-message.outputs.msg }}\n BREAKING CHANGE ${{ github.event.inputs.message }}"
-        commit_options: --amend --allow-empty
-        skip_dirty_check: true
-        push_options: --force
+        gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+        passphrase: ${{ secrets.GPG_PASSPHRASE }}
 
-    - id: get-last-commit-ref
-      run: |
-        echo "::set-output name=ref::$(git rev-parse HEAD)"
```

Comment:
Why did you delete this job ? We need it to amend the commit to trigger the release if there is a major version